CMakeLists.txt

```diff
@@ -20,29 +23,81 @@ add_compile_definitions(TFLITE_BUILD_WITH_XNNPACK_DELEGATE)
 
 # This assumes that tensorflow got checked out as submodule.
 # True, if `git clone` had the additional option `--recursive`.
-add_subdirectory(tensorflow/tensorflow/lite 
+add_subdirectory(tensorflow/tensorflow/lite
   "${CMAKE_CURRENT_BINARY_DIR}/tensorflow-lite" EXCLUDE_FROM_ALL)
 
 add_compile_definitions(DEEPSEG_VERSION=${DEEPSEG_VERSION})
 
+add_library(backscrub
+  libbackscrub.cc
+  transpose_conv_bias.cc)
+
+target_link_libraries(backscrub
+  tensorflow-lite ${CMAKE_DL_LIBS}
+  opencv_core
+  opencv_imgproc
+)
+
 add_executable(deepseg
   deepseg.cc
-  loopback.cc 
+  loopback.cc
   transpose_conv_bias.cc
 )
 
-target_link_libraries(deepseg 
-      tensorflow-lite ${CMAKE_DL_LIBS}
-      opencv_core 
-      opencv_video
-      opencv_videoio
-      opencv_imgproc
-      opencv_imgcodecs
-      opencv_highgui
+target_link_libraries(deepseg
+  backscrub
+  opencv_core
+  opencv_video
+  opencv_videoio
+  opencv_imgproc
+  opencv_imgcodecs
+  opencv_highgui
 )
 
+# Export our library, and all transitive dependencies - sadly Tensorflow Lite's
+# CMakeLists.txt does not export these for us
+function(get_link_libraries target outlist)
+  message(STATUS "get_link_libraries(${target} ${outlist})")
+  # recursive dependency munger, feed with a name and a list to extend
+  get_target_property(target_type ${target} TYPE)
+  if(${target_type} STREQUAL "INTERFACE_LIBRARY")
+    get_target_property(libs ${target} INTERFACE_LINK_LIBRARIES)
+  else()
+    get_target_property(libs ${target} LINK_LIBRARIES)
+  endif()
+  foreach(lib IN LISTS libs)
+    if(NOT TARGET ${lib})
+      continue()
+    endif()
+    get_target_property(unalias ${lib} ALIASED_TARGET)
+	if("${unalias}" STREQUAL "unalias-NOTFOUND")
+      set(unalias ${lib})
+    endif()
+	get_target_property(imp ${unalias} IMPORTED)
+	if(${imp})
+	  continue()
+	endif()
+    list(FIND ${outlist} ${unalias} exists)
+    if(NOT exists EQUAL -1)
+      continue()
+    endif()
+    list(APPEND ${outlist} ${unalias})
+    get_link_libraries(${unalias} ${outlist})
+    set(${outlist} ${${outlist}} PARENT_SCOPE)
+  endforeach()
+endfunction()
+
+set(BACKSCRUB_DEPS "")
+get_link_libraries(backscrub BACKSCRUB_DEPS)
+
+export(TARGETS
+  backscrub
+  ${BACKSCRUB_DEPS}
+  FILE BackscrubTargets.cmake)
 
 install(TARGETS deepseg)
-install(DIRECTORY models 
-    DESTINATION ${CMAKE_INSTALL_PREFIX}/share/deepbacksub 
+install(TARGETS backscrub)
+install(FILES libbackscrub.h DESTINATION include)
```

Comment:
Maybe use a subdirectory like `/usr/include/backscrub` to avoid polluting the main `/usr/include` directory with our files.

Reply:
Probably a safer option, this comes from existing CMake (and I've never actually done an install :smile: )